### PR TITLE
feat: support bounded structs 

### PR DESCRIPTION
**Pull request by wischli**
[Mashnet node companion](https://github.com/KILTprotocol/mashnet-node/pull/228)

* Fixes build issue of #13 which did not catch my attention (see CI)
* Bumps TS and Polkadot types versions

### CODE REVIEW

**Comment by wischli** [comment]
> I may be wrong, but I'd expect this to result in a type for the DidUpdate where you cannot actually set any keys or services when selecting `Change`, but instead you are faced with a value-less enum.

I suppose you are right.

Screenshot 2021-08-11 at 12 07 17

**Comment by rflechtner** [comment]
This is what we would want: 
image

**Comment by wischli** [comment]
> I may be wrong, but I'd expect this to result in a type for the DidUpdate where you cannot actually set any keys or services when selecting `Change`, but instead you are faced with a value-less enum.
> What should definitely work is the following, but that's not pretty:
> 
> ```ts
>   // We need to specify the two variants of the generic DidFragmentUpdateAction separately
>   DidFragmentUpdateAction_DidVerificationKey: {
>     _enum: {
>       Ignore: "Null",
>       Change: "DidVerificationKey",
>       Delete: "Null",
>     },
>   },
>   DidFragmentUpdateAction_ServiceEndPoints: {
>     _enum: {
>       Ignore: "Null",
>       Change: "ServiceEndPoints",
>       Delete: "Null",
>     },
>   },
> 
>   ...
>  
>   DidUpdateDetails: {
>     newAuthenticationKey: "Option<DidVerificationKey>",
>     newKeyAgreementKeys: "BTreeSet<DidEncryptionKey>",
>     attestationKeyUpdate: "DidFragmentUpdateAction_DidVerificationKey",
>     delegationKeyUpdate: "DidFragmentUpdateAction_DidVerificationKey",
>     publicKeysToRemove: "BTreeSet<KeyIdOf>",
>     serviceEndpointsUpdate: "DidFragmentUpdateAction_ServiceEndpoints",
>   },
> ```

I tried to find a solution with generics but nothing worked. I pushed your suggestion for now.

Could you do a review @rflechtner? I would like to do a runtime upgrade on Peregrine Stg and Prod tomorrow but need the types to be published before.

File: src/index.ts [hunk 1]
```diff
@@ -558,21 +558,23 @@ export const types19: RegistryTypes = {
     urls: "Vec<Url>",
     contentType: "ContentType",
   },
-  DidFragmentUpdateAction: {
+  DidFragmentUpdateAction_ServicEndpoints: {
```
[the remark this answers is not in the document]
Comment (wischli) [reply src/index.ts]: Good catch. `StakingStorageVersion` was also missing 🥲 Fixed ind fa98fac.

**Review by rflechtner — APPROVED** [review]

**Pull request merged**